mlrun/datastore/targets.py

```diff
@@ -1496,13 +1500,15 @@ def __init__(
                         f"Collection named {collection_name} is not exist in {db_name} database"
                     )
             elif override_collection:
+                # override all the collection's documents
                 db[collection_name].delete_many({})
             attr = {
                 "collection_name": collection_name,
                 "db_name": db_name,
                 "connection_string": connection_string,
             }
             path = f"mdb:///{connection_string}///{db_name}///{collection_name}"
+            mongodb_client.close()
```

Comment:
sweet 🤙 

But, I see you open and close multiple times, think if it's better to have the session open and closed once at the time the class is created and deleted (via overriding the `__del__` method). I'm not familliar with the logic of target so its just a note for you, not a thing to necessarily change.

Reply:
I think it is better like this because every time I open a connection I closed it in the end of the same method. @guy1992l

Reply:
consult with @gtopper on this one and let's hear his opinion as well :) Or maybe in the pymongo examples this is the way they recommend to work?

Reply:
@guy1992l they recommended to close after each use, and also we have few classes that can read from the same mongo data base, so it is safer to close when it possible.